File: app/src/main/java/com/zillennium/utswap/screens/navbar/walletTab/WalletFragment.java
package com.zillennium.utswap.screens.navbar.walletTab;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.zillennium.utswap.R;
import com.zillennium.utswap.bases.BaseListViewHeightActivity;
import com.zillennium.utswap.databinding.FragmentNavbarWalletBinding;
import com.zillennium.utswap.models.ListMenu.ListMenu;
import com.zillennium.utswap.models.ListMenu.ListMenuAdapter;
import com.zillennium.utswap.screens.setting.MyWallet.MyWalletActivity;
import com.zillennium.utswap.screens.setting.accountTypeScreen.AccountTypeActivity;
import com.zillennium.utswap.screens.setting.balanceScreen.BalanceActivity;
import com.zillennium.utswap.screens.setting.depositScreen.DepositActivity;
import com.zillennium.utswap.screens.setting.historicalScreen.HistoricalActivity;
import com.zillennium.utswap.screens.setting.kycScreen.KYCActivity;
import com.zillennium.utswap.screens.setting.lockUpScreen.LockUpActivity;
import com.zillennium.utswap.screens.setting.profileQRCodeScreen.ProfileQRCodeActivity;
import com.zillennium.utswap.screens.setting.profileScreen.ProfileActivity;
import com.zillennium.utswap.screens.setting.settingScreen.SettingActivity;
import com.zillennium.utswap.screens.setting.subScriptionScreen.SubscriptionActivity;
import com.zillennium.utswap.screens.setting.termScreen.TermActivity;
import com.zillennium.utswap.screens.setting.transferScreen.TransferActivity;
import com.zillennium.utswap.screens.setting.withdrawalScreen.WithdrawalActivity;

import java.util.ArrayList;

public class WalletFragment extends Fragment {

    private FragmentNavbarWalletBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        WalletViewModel walletViewModel =
                new ViewModelProvider(this).get(WalletViewModel.class);

        binding = FragmentNavbarWalletBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        // Set Activity Account
        //Name Account
        TextView txtAccount = binding.txtAccount;
        txtAccount.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), ProfileActivity.class);
            startActivity(intent);
        });

        // Profile QR Code
        LinearLayout layQrcode = binding.layQrcode;
        layQrcode.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), ProfileQRCodeActivity.class);
            startActivity(intent);
        });

        // Picture Account
        com.google.android.material.imageview.ShapeableImageView ivProfile = binding.ivProfile;
        ivProfile.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), ProfileActivity.class);
            startActivity(intent);
        });
        // Account Type
        TextView txtAccountType = binding.txtAccountType;
        txtAccountType.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), AccountTypeActivity.class);
            startActivity(intent);
        });

        // Verify
        LinearLayout txtVerify = binding.btnVerify;
        txtVerify.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), KYCActivity.class);
            startActivity(intent);
        });

        // Set Activity Setting
        ImageView btnSetting = binding.btnSetting;
        btnSetting.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), SettingActivity.class);
            startActivity(intent);
        });

        LinearLayout layWallet = binding.layWallet;
        layWallet.setOnClickListener(view -> {
            Intent intent = new Intent(getActivity(), MyWalletActivity.class);
            startActivity(intent);
        });

        // Set Balance ListView
//        int[] icon_balance = {R.drawable.ic_tasks_alt_regular,
//                R.drawable.ic_share_regular,
//                R.drawable.ic_reply_regular,
//                R.drawable.ic_paper_plane_regular,
//                R.drawable.ic_hourglass_empty_regular,
//                R.drawable.ic_lock_keyhole_regular,
//                R.drawable.ic_cart_plus_regular};
//        String[] title_balance = {"Balance",
//                "Deposit",
//                "Withdrawal",
//                "Transfer",
//                "Subscription",
//                "Lock-up",
//                "Historical"};
        int[] icon_balance = {
                R.drawable.ic_hourglass_empty_regular,
                R.drawable.ic_lock_keyhole_regular,
                R.drawable.ic_cart_plus_regular};
        String[] title_balance = {
                "Subscription",
                "Lock-up",
                "Historical"};

        ArrayList<ListMenu> accountArrayList = new ArrayList<>();

        for(int i=0; i<title_balance.length; i++){
            ListMenu account = new ListMenu(title_balance[i], icon_balance[i]);
            accountArrayList.add(account);
        }

        ListMenuAdapter accountAdapter = new ListMenuAdapter(getActivity(), accountArrayList);

        ListView lvBalance = binding.lvBalance;
        lvBalance.setAdapter(accountAdapter);
        lvBalance.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
                Intent intent;

                String title = title_balance[position];

                switch (title){
                    case "Deposit":
                        intent = new Intent(view.getContext(), DepositActivity.class);
                        break;
                    case "Withdrawal":
                        intent = new Intent(view.getContext(), WithdrawalActivity.class);
                        break;
                    case "Transfer":
                        intent = new Intent(view.getContext(), TransferActivity.class);
                        break;
                    case "Subscription":
                        intent = new Intent(view.getContext(), SubscriptionActivity.class);
                        break;
                    case "Lock-up":
                        intent = new Intent(view.getContext(), LockUpActivity.class);
                        break;
                    case "Historical":
                        intent = new Intent(view.getContext(), HistoricalActivity.class);
                        break;
                    default: intent = new Intent(view.getContext(), BalanceActivity.class); // As Balance
                }

                startActivity(intent);
            }
        });
        BaseListViewHeightActivity.setListViewHeightBasedOnItems(lvBalance);

        // Set Terms ListView
        int[] icon_terms = {R.drawable.ic_gem_regular,
                R.drawable.ic_gem_regular,
                R.drawable.ic_gem_regular};
        String[] title_terms = {"Term and Conditions",
                "Privacy Policy",
                "Operation Rules"};

        ArrayList<ListMenu> termsArrayList = new ArrayList<>();

        for(int i=0; i<title_terms.length; i++){
            ListMenu term = new ListMenu(title_terms[i], icon_terms[i]);
            termsArrayList.add(term);
        }

        ListMenuAdapter termsAdapter = new ListMenuAdapter(getActivity(), termsArrayList);

        ListView lvTerm = binding.lvTerm;
        lvTerm.setAdapter(termsAdapter);
        lvTerm.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> listView, View view, int position, long id) {
                Intent intent = new Intent(getActivity(), TermActivity.class);

                String title = title_terms[position];

                intent.putExtra("title", title);
                startActivity(intent);
            }
        });

        BaseListViewHeightActivity.setListViewHeightBasedOnItems(lvTerm);

        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/zillennium/utswap/screens/navbar/walletTab/WalletFragment.kt
package com.zillennium.utswap.screens.navbar.walletTab

import android.content.Intent
import android.widget.AdapterView.OnItemClickListener
import com.zillennium.utswap.R
import com.zillennium.utswap.UTSwapApp
import com.zillennium.utswap.bases.BaseListViewHeightActivity
import com.zillennium.utswap.bases.mvp.BaseMvpFragment
import com.zillennium.utswap.databinding.FragmentNavbarWalletBinding
import com.zillennium.utswap.models.ListMenu.ListMenu
import com.zillennium.utswap.models.ListMenu.ListMenuAdapter
import com.zillennium.utswap.screens.setting.kyc.KYCActivity
import com.zillennium.utswap.screens.wallet.transactionScreen.TransactionActivity
import com.zillennium.utswap.screens.setting.accountTypeScreen.AccountTypeActivity
import com.zillennium.utswap.screens.setting.balanceScreen.BalanceActivity
import com.zillennium.utswap.screens.wallet.depositScreen.DepositActivity
import com.zillennium.utswap.screens.wallet.historicalScreen.HistoricalActivity
import com.zillennium.utswap.screens.wallet.lockUpScreen.LockUpActivity
import com.zillennium.utswap.screens.setting.profileQRCodeScreen.ProfileQRCodeActivity
import com.zillennium.utswap.screens.setting.profileScreen.ProfileActivity
import com.zillennium.utswap.screens.setting.settingScreen.SettingActivity
import com.zillennium.utswap.screens.wallet.subScriptionScreen.SubscriptionActivity
import com.zillennium.utswap.screens.privacy.termScreen.TermActivity
import com.zillennium.utswap.screens.wallet.transferScreen.TransferActivity
import com.zillennium.utswap.screens.wallet.withdrawalScreen.WithdrawalActivity

class WalletFragment :
    BaseMvpFragment<WalletView.View, WalletView.Presenter, FragmentNavbarWalletBinding>(),
    WalletView.View {

    override var mPresenter: WalletView.Presenter = WalletPresenter()
    override val layoutResource: Int = R.layout.fragment_navbar_wallet

    override fun initView() {
        super.initView()
        try {
            binding.apply {

                // Set Activity Account
                //Name Account
                txtAccount.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, ProfileActivity::class.java)
                    startActivity(intent)
                }

                // Profile QR Code
                layQrcode.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, ProfileQRCodeActivity::class.java)
                    startActivity(intent)
                }

                // Picture Account
                ivProfile.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, ProfileActivity::class.java)
                    startActivity(intent)
                }

                // Account Type
                txtAccountType.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, AccountTypeActivity::class.java)
                    startActivity(intent)
                }

                // Verify
                btnVerify.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, KYCActivity::class.java)
                    startActivity(intent)
                }

                // Set Activity Setting
                btnSetting.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, SettingActivity::class.java)
                    startActivity(intent)
                }

                // Layout Wallet
                layWallet.setOnClickListener {
                    val intent = Intent(UTSwapApp.instance, TransactionActivity::class.java)
                    startActivity(intent)
                }

                // Set Balance ListView
//        int[] icon_balance = {R.drawable.ic_tasks_alt_regular,
//                R.drawable.ic_share_regular,
//                R.drawable.ic_reply_regular,
//                R.drawable.ic_paper_plane_regular,
//                R.drawable.ic_hourglass_empty_regular,
//                R.drawable.ic_lock_keyhole_regular,
//                R.drawable.ic_cart_plus_regular};
//        String[] title_balance = {"Balance",
//                "Deposit",
//                "Withdrawal",
//                "Transfer",
//                "Subscription",
//                "Lock-up",
//                "Historical"};

                // Set Balance ListView
//        int[] icon_balance = {R.drawable.ic_tasks_alt_regular,
//                R.drawable.ic_share_regular,
//                R.drawable.ic_reply_regular,
//                R.drawable.ic_paper_plane_regular,
//                R.drawable.ic_hourglass_empty_regular,
//                R.drawable.ic_lock_keyhole_regular,
//                R.drawable.ic_cart_plus_regular};
//        String[] title_balance = {"Balance",
//                "Deposit",
//                "Withdrawal",
//                "Transfer",
//                "Subscription",
//                "Lock-up",
//                "Historical"};
                val iconBalance = intArrayOf(
                    R.drawable.ic_hourglass_empty_regular,
                    R.drawable.ic_lock_keyhole_regular,
                    R.drawable.ic_cart_plus_regular
                )
                val titleBalance = arrayOf(
                    "Subscription",
                    "Lock-up",
                    "Historical"
                )

                val accountArrayList = ArrayList<ListMenu>()

                for (i in titleBalance.indices) {
                    val account = ListMenu(titleBalance[i], iconBalance[i])
                    accountArrayList.add(account)
                }

                val accountAdapter = ListMenuAdapter(UTSwapApp.instance, accountArrayList)

                val lvBalance = binding.lvBalance
                lvBalance.adapter = accountAdapter
                lvBalance.onItemClickListener =
                    OnItemClickListener { _, _, position, _ ->
                        val intent: Intent
                        val title = titleBalance[position]
                        intent = when (title) {
                            "Deposit" -> Intent(UTSwapApp.instance, DepositActivity::class.java)
                            "Withdrawal" -> Intent(UTSwapApp.instance, WithdrawalActivity::class.java)
                            "Transfer" -> Intent(UTSwapApp.instance, TransferActivity::class.java)
                            "Subscription" -> Intent(UTSwapApp.instance, SubscriptionActivity::class.java)
                            "Lock-up" -> Intent(UTSwapApp.instance, LockUpActivity::class.java)
                            "Historical" -> Intent(UTSwapApp.instance, HistoricalActivity::class.java)
                            else -> Intent(UTSwapApp.instance, BalanceActivity::class.java) // As Balance
                        }
                        startActivity(intent)
                    }
                BaseListViewHeightActivity.setListViewHeightBasedOnItems(lvBalance)

                // Set Terms ListView
                val iconTerms = intArrayOf(
                    R.drawable.ic_gem_regular,
                    R.drawable.ic_gem_regular,
                    R.drawable.ic_gem_regular
                )
                val titleTerms = arrayOf(
                    "Term and Conditions",
                    "Privacy Policy",
                    "Operation Rules"
                )

                val termsArrayList = ArrayList<ListMenu>()

                for (i in titleTerms.indices) {
                    val term = ListMenu(titleTerms[i], iconTerms[i])
                    termsArrayList.add(term)
                }

                val termsAdapter = ListMenuAdapter(UTSwapApp.instance, termsArrayList)

                lvTerm.adapter = termsAdapter
                lvTerm.onItemClickListener =
                    OnItemClickListener { _, _, position, _ ->
                        val intent = Intent(UTSwapApp.instance, TermActivity::class.java)
                        val title = titleTerms[position]
                        intent.putExtra("title", title)
                        startActivity(intent)
                    }

                BaseListViewHeightActivity.setListViewHeightBasedOnItems(lvTerm)

            }

        } catch (error: Exception) {
            // Must be safe
        }
    }
}